credentials/src/main/kotlin/web5/credentials/VerifiableCredential.kt

```diff
@@ -0,0 +1,275 @@
+ *
+ * @property vcDataModel The [VcDataModel] instance representing the core data model of a verifiable credential.
+ */
+public class VerifiableCredential(private val vcDataModel: VcDataModel) {
+  /**
+   * Sign a verifiable credential using a specified decentralized identifier ([did]) and an optional key alias ([keyAlias]).
+   *
+   * If the [keyAlias] is null, the function will attempt to use the first available verification method from the [did].
+   * The result is a String in a JWT format.
+   *
+   * @param did The [Did] used to sign the credential.
+   * @param keyAlias An optional alias for the key used to sign the credential.
+   * @return The JWT representing the signed verifiable credential.
+   *
+   * Example:
+   * ```
+   * val signedVc = verifiableCredential.sign(myDid)
+   * ```
+   */
+  public fun sign(did: Did, keyAlias: String? = null): String {
+    val keyAliaz = keyAlias ?: run {
+      val didResolutionResult = DidResolvers.resolve(did.uri)
+      val verificationMethod = didResolutionResult.didDocument.allVerificationMethods[0]
+
+      require(verificationMethod != null) { "no key alias found" }
+
+      verificationMethod.id.toString()
+    }
+
+    // TODO: figure out how to make more reliable since algorithm is technically not a required property of a JWK
+    // BUT we always include it in our JWKs
+    val publicKey = did.keyManager.getPublicKey(keyAliaz)
+    val algorithm = publicKey.algorithm
+    val jwsAlgorithm = JWSAlgorithm.parse(algorithm.toString())
+
+    val jwtHeader = JWSHeader.Builder(jwsAlgorithm)
+      .keyID(keyAlias)
+      .build()
+
+    val jwtPayload = JWTClaimsSet.Builder()
+      .subject(vcDataModel.credentialSubject.id.toString())
+      .claim("vc", vcDataModel.toMap())
+      .build()
+
+    val jwtObject = SignedJWT(jwtHeader, jwtPayload)
+    val toSign = jwtObject.signingInput
+
+    val signatureBytes = did.keyManager.sign(keyAliaz, toSign)
+
+    val base64UrlEncodedSignature = Base64URL(Convert(signatureBytes).toBase64Url(padding = false))
+    val base64UrlEncodedHeader = jwtHeader.toBase64URL()
+    val base64UrlEncodedPayload = jwtPayload.toPayload().toBase64URL()
+
+    return "$base64UrlEncodedHeader.$base64UrlEncodedPayload.$base64UrlEncodedSignature"
+  }
+
+  /**
+   * Checks if the given [presentationDefinition] is satisfied based on the provided input descriptors and constraints.
+   *
+   * @param presentationDefinition The Presentation Definition to be evaluated.
+   * @return `true` if the Presentation Definition is satisfied, `false` otherwise.
+   * @throws NotImplementedError if certain features like Submission Requirements or Field Filters are not implemented.
+   */
+  public fun satisfiesPresentationDefinition(presentationDefinition: PresentationDefinitionV2): Boolean {
+    if (!presentationDefinition.submissionRequirements.isNullOrEmpty()) {
+      throw NotImplementedError("Presentation Definition's Submission Requirements feature is not implemented")
```

Comment:
`Error` typically is a serious condition that is unrecoverable. Perhaps consider `UnsupportedOperationException` or build a custom type that is analogous to "UnsupportedDataException".

Reply:
considered: https://github.com/TBD54566975/web5-kt/commit/e3458b8e31bc9cd3a270bdd6026776b5c68ca190#diff-1bba7d44ff1f4bc8be5a1fe55d709b41263647f67d9da9cb16cde7a1407e7922